### PR TITLE
Added the functionality to convert dom::object and dom::array to dom element.

### PR DESCRIPTION
**Pull request by didarpin**
closed: #2218 

Added the functionality to convert dom::object and dom::array to dom element.

Add the function "inline operator element() const noexcept" to the dom::object and dom::array classes, and add test for it.

### CODE REVIEW

**Comment by lemire** [comment]
Running tests.

File: tests/dom/basictests.cpp [hunk 1]
```diff
@@ -1351,6 +1381,8 @@ namespace dom_api_tests {
            array_iterator_empty() &&
            object_iterator_advance() &&
            array_iterator_advance() &&
+		   convert_object_to_element() &&
```
Comment (lemire) [inline tests/dom/basictests.cpp:1384]: Here the new code is not indented like the previous and following lines. We do not enforce style or strict rules, but it should be trivial to fix...

Comment (didarpin) [reply tests/dom/basictests.cpp]: I'm sorry I didn't catch this issue sooner. I've fixed it now.

Comment (lemire) [reply tests/dom/basictests.cpp]: That's fine. We do not have rules to that effect so your code was fine.

**Comment by lemire** [comment]
Running tests. I expect to merge.

**Comment by lemire** [comment]
This will part of the next release.

**Comment by didarpin** [comment]
That's great!

**Pull request merged**